go/fn/resourcelist.go

```diff
@@ -195,42 +195,68 @@ func (rl *ResourceList) UpsertObjectToItems(obj interface{}, checkExistence func
 	return nil
 }
 
+func (rl *ResourceList) LogResult(err error) {
+	// If the error is not a Results type, we wrap the error as a Result.
+	if err == nil {
+		return
+	}
+	if results, ok := err.(Results); ok {
```

Comment:
I guess we should also check for type `ErrOpOrDie`.

Reply:
I think the `default` case will handle this correctly:
```
default:
  rl.Results = append(rl.Results, ErrorResult(err))
```

Reply:
I thought we can do something with the `fields` field in `ErrOpOrDie`. But I just found they are private. So I guess we can use default to handle it.